### PR TITLE
add a command interface to generate/clear time barriers

### PR DESCRIPTION
**Pull request by phlptp**
fix the command interface in the C++98 API
update docs
add query to get the current barriers

### Summary

If merged this pull request will

- add a command interface to generate/clear time barriers
- fix the command interface in the C++98 API
- update docs
- add query to get the current barriers

### CODE REVIEW

**Review by trevorhardy — APPROVED** [review]
I/we need to add to the command documentation but this is a great start; we can handle this in another PR.

File: docs/user-guide/advanced_topics/commandInterface.md [hunk 1]
```diff
@@ -50,27 +50,33 @@ The `commandStr` is a generic string, so can be anything that can be contained i
...
+| ``logbuffer <size>``      | [all objects] set the log buffer to a particular size or `stop`                                |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``monitor <args...>``     | [brokers] set up a federate as the time monitor <args...> = <federate names> <logperiod>       |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``set barrier <args...>`` | [broker,federates] set a time barrier <args...> = <time> <barrier_id*>                         |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``clear barrier <id*>``   | [broker,federates] clear a time barrier <barrier_id*>                         |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``remotelog <level>``     | [all object] instruct the object to send log messages to a remote location in addition to local|
```
Comment (nightlark) [inline docs/user-guide/advanced_topics/commandInterface.md:70]:
```suggestion
| ``remotelog <level>``     | [all objects] instruct the object to send log messages to a remote location in addition to local|
```

File: docs/user-guide/advanced_topics/commandInterface.md [hunk 2]
```diff
@@ -50,27 +50,33 @@ The `commandStr` is a generic string, so can be anything that can be contained i
...
+| ``echo``                  | [all objects] send a command with a `commandStr`=`echo_reply` back to the sender               |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``log <string>``          | [all objects] generate a log message in a particular object                                    |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``logbuffer <size>``      | [all objects] set the log buffer to a particular size or `stop`                                |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``monitor <args...>``     | [brokers] set up a federate as the time monitor <args...> = <federate names> <logperiod>       |
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``set barrier <args...>`` | [broker,federates] set a time barrier <args...> = <time> <barrier_id*>                         |
```
Comment (nightlark) [inline docs/user-guide/advanced_topics/commandInterface.md:66]:
```suggestion
| ``set barrier <args...>`` | [brokers,federates] set a time barrier <args...> = <time> <barrier_id*>                         |
```

File: docs/user-guide/advanced_topics/commandInterface.md [hunk 3]
```diff
@@ -50,27 +50,33 @@ The `commandStr` is a generic string, so can be anything that can be contained i
...
++---------------------------+------------------------------------------------------------------------------------------------+
+| ``clear barrier <id*>``   | [broker,federates] clear a time barrier <barrier_id*>                         |
```
Comment (nightlark) [inline docs/user-guide/advanced_topics/commandInterface.md:68]:
```suggestion
| ``clear barrier <id*>``   | [brokers,federates] clear a time barrier <barrier_id*>                         |
```

File: src/helics/core/CommonCore.cpp [hunk 4]
```diff
@@ -2426,15 +2426,15 @@ void CommonCore::setIdentifier(std::string_view name)
...
-};
+static const std::map<std::string_view, std::pair<std::uint16_t, QueryReuse>> mapIndex{
+    {"global_time", {CURRENT_TIME_MAP, QueryReuse::DISABLED}},
+    {"global_status", {GLOBAL_STATUS, QueryReuse::DISABLED}},
+    {"dependency_graph", {DEPENDENCY_GRAPH, QueryReuse::ENABLED}},
+    {"data_flow_graph", {DATA_FLOW_GRAPH, QueryReuse::ENABLED}},
+    {"barriers", {BARRIERS, QueryReuse::DISABLED}},
+    {"global_state", {GLOBAL_STATE, QueryReuse::DISABLED}},
+    {"global_time_debugging", {GLOBAL_TIME_DEBUGGING, QueryReuse::DISABLED}},
+    {"global_flush", {GLOBAL_FLUSH, QueryReuse::DISABLED}}};
```
Comment (nightlark) [inline src/helics/core/CommonCore.cpp:2437]: It seems like the QueryReuse value mapping of the std::pair is different after the change from a bool to QueryReuse -- is that intentional?

Specifically, if true -> QueryReuse::DISABLED and false -> queryReuse::Enabled, then I'd expect global_status to be QueryReuse::ENABLED.

File: src/helics/core/queryHelpers.hpp [hunk 5]
```diff
@@ -30,9 +30,13 @@ enum Subqueries : std::uint16_t {
     GLOBAL_STATE = 6,
     GLOBAL_TIME_DEBUGGING = 7,
     GLOBAL_FLUSH = 8,
-    GLOBAL_STATUS = 9
+    GLOBAL_STATUS = 9,
+    BARRIERS = 11
 };
 
+/// Enumeration is query result is reusable
```
Comment (nightlark) [inline src/helics/core/queryHelpers.hpp:37]:
```suggestion
/// Enumeration of if query result is reusable
```

File: src/helics/cpp98/Federate.hpp [hunk 6]
```diff
@@ -617,6 +604,52 @@ class Federate {
         return result;
     }
 
+    /**
+     * Send a command to another HELICS object through a federate.
+     *
+     * @param target The name of the object to send the command to.
+     * @param command The command to send.
```
Comment (nightlark) [inline src/helics/cpp98/Federate.hpp:611]:
```suggestion
     * @param cmd The command to send.
```

File: src/helics/cpp98/Federate.hpp [hunk 7]
```diff
@@ -617,6 +604,52 @@ class Federate {
...
+     */
+    const char* getCommand()
+    {
+        return helicsFederateGetCommand(fed, HELICS_IGNORE_ERROR);
+    }
+
+    /**
+     * Get a command sent to the federate. Blocks until a command is received.
+     *
+     * @param fed The federate to get the command for.
```
Comment (nightlark) [inline src/helics/cpp98/Federate.hpp:633]:
```suggestion
```

File: src/helics/core/CommonCore.cpp (hunk 4 above)
Comment (phlptp) [reply src/helics/core/CommonCore.cpp:2437]: Yes,  I realized that the previous value for global_status was not correct, that query can change and should be reset.    So was probably a bug.

**Review by nightlark — APPROVED** [review]

**Pull request merged**